language/expressions.md

```diff
@@ -720,9 +726,26 @@ result. Chained assignments are permitted.
 
 Examples:
 
+~~~puppet
+
     $a = 10
     $x = $y = 0
 
+    # from array
+    [$a, $b] = [1, 2]  # assigns 1 to $a, and 2 to $b
+
+    # from hash
+    [$a, $b] = { a => 10, b => 20, c => 30 }  # assigns 10 to $a, and 20 to $b
+
+    # from class
+    class mymodule::someclass::example($x = 100) {
+      $a = 10
+    }
+    include example
+    [$a, $x] = Class['mymodule::someclass::example']  # assigns 10 to $a, and 100 to $x
```

Comment:
Yeah, I see the use for sure, final docs should probably just be clear to call out that this will export all variables including private ones.

And I would think something like this might be handy in wrapper class type situation where you'd extract all the expected parameters from another class or define, except you can't because you also get internal stuff?

Are you looking to add this to resources as well, including defines?